src/bodywork/k8s/workflow_jobs.py

```diff
@@ -185,6 +183,70 @@ def create_workflow_cronjob(cron_job: k8s.V1Job) -> None:
     )
 
 
+def update_workflow_cronjob(
+    namespace: str,
+    name: str,
+    schedule: str = None,
+    project_repo_url: str = None,
+    project_repo_branch: str = None,
+    retries: int = None,
+    successful_jobs_history_limit: int = None,
+    failed_jobs_history_limit: int = None,
+) -> None:
+    """Update a Bodywork workflow cronjob.
+
+    :param name: The name  of the cronjob.
+    :param namespace: Namespace the cronjob resides in.
+    :param schedule: A valid cron schedule definition.
+    :param project_repo_url: The URL for the Bodywork project Git
+        repository.
+    :param project_repo_branch: The Bodywork project Git repository
+        branch to use, defaults to 'master'.
+    :param retries: Number of times to retry running the stage to
+        completion (if necessary), defaults to 2.
+    :param successful_jobs_history_limit: The number of successful job
+        runs (pods) to keep, defaults to 1.
+    :param failed_jobs_history_limit: The number of unsuccessful job
+        runs (pods) to keep, defaults to 1.
+    """
+
+    if not schedule:
+        schedule = (
+            k8s.BatchV1beta1Api()
+            .read_namespaced_cron_job(name, namespace)
+            .spec.schedule
+        )
+
+    if project_repo_url and project_repo_branch:
+        pod_spec = k8s.V1PodSpec(
+            containers=[
+                k8s.V1Container(
+                    name="bodywork", args=[project_repo_url, project_repo_branch]
+                )
+            ],
+        )
+        job = k8s.V1Job(
+            spec=k8s.V1JobSpec(
+                template=k8s.V1PodTemplateSpec(spec=pod_spec),
+                backoff_limit=retries,
+            )
+        )
+        job_template = k8s.V1beta1JobTemplateSpec(spec=job.spec)
+    else:
+        job_template = k8s.V1beta1JobTemplateSpec()
+
+    cronjob = k8s.V1beta1CronJob(
+        spec=k8s.V1beta1CronJobSpec(
+            job_template=job_template,
+            schedule=schedule,
+            successful_jobs_history_limit=successful_jobs_history_limit,
```

Comment:
Should we not need to retrieve `successful_jobs_history_limit` and `failed_jobs_history_limit` if they're `None`?

Reply:
This is the update/patch method, if they are `None` then they are not updated. If we wrote a conditional  to omit these parameters if they are `None` then it would have no effect and therefore it's not necessary.

Reply:
By mistake - I misunderstood how the `patch` methods work.

Reply:
It took me a while to extract that information from the docs